### PR TITLE
Deprecate signed std::num::NonZeroI* with a call for use cases 

### PR DESCRIPTION
**Pull request by SimonSapin**
CC https://github.com/rust-lang/rust/issues/49137#issuecomment-375823481

### CODE REVIEW

**Comment by alexcrichton** [comment]
I think these were added a few nighties ago, right? If so it may be fine to just delete them

**Comment by SimonSapin** [comment]
Yeah I agree that’d be fine in terms of breakage, but keeping them with a warning gets us a way to specifically ask people to comment on the tracking issue. I’m not too sure that’s really useful though. What do you think?

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -93,18 +95,26 @@ macro_rules! nonzero_integers {
...
 }
 
 nonzero_integers! {
-    // Change this to `#[unstable(feature = "i128", issue = "35118")]`
-    // if other NonZero* integer types are stabilizied before 128-bit integers
     #[unstable(feature = "nonzero", issue = "49137")]
-    NonZeroU128(u128); NonZeroI128(i128);
+    #[rustc_deprecated(since = "1.26.0", reason = "\
+        signed non-zero integers are considered for removal due to lack of known use cases. \
+        If you’re using them, please comment on https://github.com/serde-rs/serde/pull/1191")]
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply src/libcore/num/mod.rs]: Oops :) Fixed, thanks.

**Comment by SimonSapin** [comment]
Discussed in person with Alex

@bors: r=alexcrichton

**Pull request merged**